app/Traits/ArclightConnection.php

```diff
@@ -2,37 +2,25 @@
 
 namespace App\Traits;
 
+use Symfony\Component\HttpFoundation\Response as HttpResponse;
+use App\Services\Arclight\ArclightService;
+
 use Exception;
 
 trait ArclightConnection
 {
     private function fetchArclight($path, $language_id = null, $include_refs = false, $parameters = '')
     {
-        $base_url = config('services.arclight.url');
-        $key      = config('services.arclight.key');
-
-        $path = $base_url.$path.'?_format=json&apiKey='.$key.'&limit=3000&platform=ios';
-
-        if ($language_id) {
-            $path .= '&languageIds='.$language_id;
-        }
-
-        if ($include_refs) {
-            $refs = implode(',', array_keys($this->getIdReferences()));
-            $path .= '&ids='.$refs;
-        }
-
-        $path .= '&'.$parameters;
+        $new_path = ArclightService::createPath($path, $language_id, $include_refs, $parameters);
 
         try {
             $ctx = stream_context_create(array('http'=>
                 array('timeout' =>  (int) config('services.arclight.service_timeout'))
             ));
-            $results = json_decode(file_get_contents($path, false, $ctx));
+            $results = json_decode(file_get_contents($new_path, false, $ctx));
         } catch (Exception $e) {
-            return strpos($e, 'timed out') !== false
-                ? $this->setStatusCode(408)->replyWithError('Request timeout')
-                : $this->setStatusCode(500)->replyWithError('Internal server error');
+            \Log::channel('errorlog')->error(["Arclight Connection Error: '{$e->getMessage()}" ]);
+            $results = [];
```

Comment:
@bradflood For now on, It will avoid return a 500 or 401 status code when arclight API returns a status code different to 200 status code.